bolt/lib/Profile/DataAggregator.cpp

```diff
@@ -2067,7 +2064,21 @@ std::error_code DataAggregator::parseMMapEvents() {
       }
     }
 
-    BinaryMMapInfo.insert(std::make_pair(MMapInfo.PID, MMapInfo));
+    // In some larger binaries, the loaded binary gets a a second text segment
```

Comment:
Essentially, instead of storing the mapping for the first segment, you suggest to store the base address and the last address calculated as `BaseAddress + Size`, right? We will need to update the description of `MMapInfo` structure accordingly.